File: Task 2/Java/app/src/main/java/dev/ogabek/java/manager/RealmManager.java
package dev.ogabek.java.manager;

import java.util.ArrayList;

import dev.ogabek.java.model.Note;
import io.realm.Realm;
import io.realm.RealmResults;

public class RealmManager {

    private static final String TAG = RealmManager.class.toString();

    private static RealmManager realmManager;
    private static Realm realm;

    public RealmManager getInstance() {
        if (realmManager == null) {
            realmManager = new RealmManager();
        }
        return realmManager;
    }

    public RealmManager() {
        realm = Realm.getDefaultInstance();
    }

    public void saveNote(Note note) {
        realm.beginTransaction();
        realm.copyToRealmOrUpdate(note);
        realm.commitTransaction();
    }

    public ArrayList<Note> getData() {
        RealmResults<Note> result = realm.where(Note.class).findAll();
        return new ArrayList<>(result);
    }

}

File: Task 2/Kotlin/app/src/main/java/dev/ogabek/kotlin/manager/RealmManager.kt
package dev.ogabek.kotlin.manager

import android.util.Log
import dev.ogabek.kotlin.model.Note
import io.realm.Realm
import io.realm.RealmResults

class RealmManager {

    private val TAG = RealmManager::class.java.simpleName

    companion object {
        var realmManager: RealmManager? = null
        private lateinit var realm: Realm
        val getInstance: RealmManager?
            get() {
                if (realmManager == null) {
                    realmManager = RealmManager()
                }
                return realmManager
            }
    }

    init {
        realm = Realm.getDefaultInstance()
    }

    fun savePost(note: Note) {
        realm.beginTransaction()
        realm.copyToRealmOrUpdate(note)
        realm.commitTransaction()
    }

    fun loadPosts(): ArrayList<Note> {
        val result: RealmResults<Note> = realm.where(Note::class.java).findAll()
        return ArrayList(result)
    }

}